### ISSUE TITLE
Fix false negatives for babel with preset "react-app"

### ISSUE THREAD

**Issue by fffed** [stylelint/stylelint#5229]
> Clearly describe the bug

There are no errors or warnings for css-in-js when `.babelrc` looks like:
```json
{
  "presets": [
    "react-app"
  ]
}
```
 while properly working for `less` files.
But when presets are empty:
```json
{
  "presets": [ ]
}
```
Errors appear as intended.

> Which rule, if any, is the bug related to?

any rule.

> What code is needed to reproduce the bug?

e.g.

```css
a {
  width : 00.1em;
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": ["stylelint-config-standard"],
}
```

> Which version of stylelint are you using?

`13.12.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

"CLI with `stylelint "**/*.jsx"`"

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

"Yes, it's related to css-in-js"

> What did you expect to happen?

"Errors to be flagged."

> What actually happened (e.g. what warnings or errors did you get)?

"No any errors"

**Comment by jeddy3** [comment]
@fffed Thanks for the report and for using the template.

It's likely a problem with [the parser](https://github.com/stylelint/postcss-css-in-js) stylelint uses [to support CSS-in-JS](https://stylelint.io/about/syntaxes). I've added this issue to https://github.com/stylelint/stylelint/issues/4574, which is a list of the problems with the syntax that we need help to fix.

Please consider [contributing to the parser](https://stylelint.io/developer-guide/syntaxes) if you have time.

**Comment by jeddy3** [comment]
Closing as the [`syntax` option was removed](https://github.com/stylelint/stylelint/issues/5289) for the [14.0.0 release](https://github.com/stylelint/stylelint/issues/5205). (See the [migration guide](https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-14.md#syntax-option-and-automatic-inferral-of-syntax).)

We'll also [be deprecating](https://github.com/stylelint/postcss-css-in-js/issues/225) our [forked CSS-in-JS package](https://github.com/stylelint/postcss-css-in-js), in favour of leaner custom syntaxes.

Please consider [writing a custom syntax](https://github.com/stylelint/stylelint/blob/main/docs/developer-guide/syntaxes.md) specific to the CSS-in-JS library in this issue and fixing the bug there.
